### DISCUSSION TITLE
Ignore buffers created by no-neck-pain.nvim

### DISCUSSION THREAD

**Discussion by okkdev** [olimorris/persisted.nvim#128] [Q&A]
Hi, I'm using [no-neck-pain.nvim](https://github.com/shortcuts/no-neck-pain.nvim). That plugin creates buffers on the sides to center your code.
Persisted autosaves these buffers too, which messes up the actual last opened file. So I'd like to ignore them completely.

I use the NoNeckPain buffers as scratch pads for notes so the buffers are files named "no-neck-pain-left.norg" and "no-neck-pain-right.norg", but are located outside of my current project folder.

So I tried adding "~/Documents/Notes" to the `ignored_dirs`, but that didn't work.

I've tried closing NoNeckPain pre save, but that didn't work either.

```lua
vim.api.nvim_create_autocmd("User", {
  pattern = "PersistedSavePre", 
  callback = function()
    require("no-neck-pain").disable()
  end
}) 
```

I also tried with the `should_autosave` setting.

```lua
should_autosave = function()
  if string.find(vim.api.nvim_buf_get_name(vim.bo), "no-neck-pain-") then
    return false
  end
  return true
end
```

I'm not sure how to get rid of these buffers. Would appreciate any ideas.

**Comment by olimorris** [comment]
`PersistedSavePre` will fire before the session is written to disk so will be calling `require("no-neck-pain").disable()`. Does the `disable()` command close those scratch buffers?

**Reply by okkdev** [reply]
Yes, I don't see them in my open buffers after `require("no-neck-pain").disable()`.

**Reply by olimorris** [reply]
Can you open up a bug report and follow the instructions in that. I can take a look.

If you do a print statement followed by a sleep do you see any output?

**Reply by okkdev** [reply]
I can do that, thanks. :)

I guess dumb question, but how do I add a sleep?

**Reply by shortcuts** [reply]
You can do `vim.loop.sleep(ms)`

Thanks for testing the plugin and filling the issue btw, let me know if I can help making the integration between NNP and persisted easier :)

**Reply by okkdev** [reply]
Thanks!

I added:
```lua
require("no-neck-pain").disable()
print("hello")
vim.loop.sleep(5000)
```

When I `:q` now, the center buffer (my code) closes, the noneckpain sides remain. The `print("hello")` doesn't appear. And it sleeps for 5 seconds. Then nvim exits.
The print might not show because I use Noice. 👀 

I'll create the report now.

**Reply by okkdev** [reply]
Bug report: https://github.com/olimorris/persisted.nvim/issues/129
